src/DssSpell.sol

```diff
@@ -23,22 +23,62 @@ import "dss-exec-lib/DssAction.sol";
 
-interface GemJoin6Like {
-    function setImplementation(
-        address implementation,
-        uint256 permitted
-    ) external;
+interface DssVestLike {
+    function create(address, uint256, uint256, uint256, uint256, address) external returns (uint256);
+    function restrict(uint256) external;
+    function yank(uint256) external;
+}
+
+interface GemLike {
+    function transfer(address, uint256) external returns (bool);
 }
 
 contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
     // Provides a descriptive tag for bot consumption
     // This should be modified weekly to provide a summary of the actions
-    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/821b3cd2d736be97bedb7583dcbff49edeb53e1f/governance/votes/Executive%20vote%20-%20March%2025%2C%202022.md -q -O - 2>/dev/null)"
+    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/a7095d5b92ee825bef28b6f5d22baec50718d438/governance/votes/Executive%20vote%20-%20April%201%2C%202022.md -q -O - 2>/dev/null)"
     string public constant override description =
-        "2022-03-25 MakerDAO Executive Spell | Hash: 0xde56d1553299f5d1dc2fdeb7ca352d82ac2ca011b9afbaa0709fe42338871d29";
+        "2022-04-01 MakerDAO Executive Spell | Hash: 0x4ac0f251ca491bf27799ebe04452ad4a6f48f2c5c8a09a5c2880a984c2f26178";
+
+    uint256 constant WAD = 10**18;
+
+    DssVestLike constant MCD_VEST_DAI = DssVestLike(0x2Cc583c0AaCDaC9e23CB601fDA8F1A0c56Cdcb71);
+    DssVestLike constant MCD_VEST_MKR_TREASURY = DssVestLike(0x6D635c8d08a1eA2F1687a5E46b666949c977B7dd);
+
+
+    // Gov Dai Transfer, Stream and MKR vesting (41.20 MKR)
+    address constant GOV_WALLET_1      = 0x01D26f8c5cC009868A4BF66E268c17B057fF7A73;
+    // Gov MKR vesting (73.70 MKR) and MKR Transfer (60 MKR)
+    address constant GOV_WALLET_2      = 0xC818Ae5f27B76b4902468C6B02Fd7a089F12c07b;
+    // Gov MKR vesting (52.74 MKR)
+    address constant GOV_WALLET_3      = 0xbfDD0E744723192f7880493b66501253C34e1241;
+    // Immunifi Core Unit
+    address constant ISCU_WALLET       = 0xd1F2eEf8576736C1EbA36920B957cd2aF07280F4;
+    // Real World Finance Core Unit
+    address constant RWF_WALLET        = 0x96d7b01Cc25B141520C717fa369844d34FF116ec;
+    // Gelato Keeper Network Contract for Dai Stream
+    address constant GELATO_WALLET     = 0x926c21602FeC84d6d0fA6450b40Edba595B5c6e4;
+
+    // Start Dates - Start of Day
+    uint256 constant FEB_08_2022 = 1644278400;
+    uint256 constant MAR_01_2022 = 1646092800;
+    uint256 constant APR_01_2022 = 1648771200;
+
+    // End Dates - End of Day
+    uint256 constant ONE_YEAR = 365 days;
+    // 2022-03-01 to 2022-08-01
+    uint256 constant FIVE_MONTHS = 153 days;
+    // 2022-04-01 to 2022-10-01
+    uint256 constant SIX_MONTHS = 183 days;
+    // 2022-04-01 to 2022-12-31
+    uint256 constant EIGHT_MONTHS = 274 days;
```

Comment:
BTW, validated all durations are correct.  I have lots of fingers and toes.

Reply:
I used this: https://www.epochconverter.com/ for the calc since it goes both ways. I liked Brian's idea once i understood it, but i agree it is making this one weird.  will revert to end dates.  To make sure i do it right: the end date should be at 00:00:00 UTC on the date specified? so 2022-03-01- 2022-08-01 would be in effect from 2022-03-01 00:00:00 UTC to 2022-07-31 23:59:59